inductiva/tasks/task.py

```diff
@@ -379,18 +381,29 @@ def download_outputs(
             rm_downloaded_zip_archive: Whether to remove the archive after
             uncompressing it. If uncompress is False, this argument is ignored.
         """
-        api_response = self._api.download_task_output(
-            path_params=self._get_path_params(),
-            query_params={
-                "filename": filenames or [],
-            },
-            stream=True,
-            skip_deserialization=True,
-        )
-        # use raw urllib3 response instead of the generated client response, to
-        # implement our own download logic (with progress bar, first checking
-        # the size of the file, etc.)
-        response = api_response.response
+        api_response = self._api.get_output_download_url(
```

Comment:
Thanks Bruno, this download_outputs function is becoming a bit too complex with lots of conditional logic, do you think you can simplify it a bit? I will leave a couple of suggestions...